prac_05/wimbledon.py

```diff
@@ -0,0 +1,43 @@
+def read_wimbledon_data(filename):
+    champions = {}  # Dictionary to store champions and their win counts
+    countries = set()  # Set to store countries of champions
+
+    with open(filename, "r", encoding="utf-8-sig") as in_file:
+        # Skip the header line
+        next(in_file)
+
+        for line in in_file:
+            data = line.strip().split(",")
+            champion = data[2]
+            country = data[1]
+
+            if champion in champions:
+                champions[champion] += 1
+            else:
+                champions[champion] = 1
+
+            countries.add(country)
+
+    return champions, sorted(countries)
+
+
+def display_champions(champions):
+    print("Wimbledon Champions:")
+    for champion, wins in champions.items():
+        print(f"{champion} {wins}")
+
+
+def display_countries(countries):
+    print("\nThese countries have won Wimbledon:")
+    countries_string = ", ".join(countries)
+    print(countries_string)
+
+
+def main():
+    filename = "wimbledon.csv"
+    champions, countries = read_wimbledon_data(filename)
+    display_champions(champions)
+    display_countries(countries)
```

Comment:
This main structure is very easy to read, maybe you could've combined your display functions to make display results? Either way, achieves the same thing. Main function is at the bottom and usually belongs at the top of the file. filename = "wimbledon.csv" looks like a constant, FILENAME outside of the main function might be what you want here.

Reply:
Thank you, I did consider combining the two display functions however in the task it does say 4 functions including main and I was not sure what the other would be. Lindsay does also talk about how functions should only do 1 thing so I thought that could be him testing to make sure that is kept to. As for the other suggestions I do think they are good to implement and have done so thank you.

Reply:
4 is not a special goal. Mine has one print function. Load, process, are the other two.